aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -90,13 +125,37 @@ struct AT_CUDA_API CUDAEvent {
 
   // Note: cudaStreamWaitEvent must be called on the same device as the stream.
   // The event has no actual GPU resources associated with it.
-  void block(const CUDAStream& stream) {
+  void block(const CUDAStream& stream) const {
     if (is_created_) {
       CUDAGuard guard(static_cast<int16_t>(stream.device_index()));
       AT_CUDA_CHECK(cudaStreamWaitEvent(stream, event_, 0));
     }
   }
 
+  float elapsed_time(const CUDAEvent& other) const {
+    float time_ms = 0;
+    // raise cudaErrorNotReady if either event is recorded but not yet completed
```

Comment:
I think this needs to check that `this` and `other` were created. Otherwise, the `event_` fields may have undefined values.